[13.0][ADD] storage_backend_ftp

acsonefho opened the pull request:
New module `storage_backend_ftp`

**Note:** I didn't test every security protocols

acsonefho committed:
[13.0][ADD] storage_backend_ftp

LoisRForgeFlow commented on storage_backend_ftp/components/ftp_adapter.py:
```diff
@@ -0,0 +1,149 @@
+        full_path = self._fullpath(relative_path)
+        with ftp(self.collection) as client:
+            try:
+                return client.retrlines("LIST " + full_path)
```
This does not return a list. But a string with a command-like message. I think you should use `client.nlst(full_path)`

LoisRForgeFlow replied:
addressed in https://github.com/acsone/storage/pull/1.

LoisRForgeFlow reviewed (commented):
I did an initial test, see a comment inline.

LoisRForgeFlow committed:
[FIX] storage_backend_ftp: fix list and get
* list should return a python list of files.
* get should get and directly return the bynary from the FTP.
  Before, a string was being retrived and then encoded. This is
  problematic if the original binary was not utf-8.

LoisRForgeFlow commented:
Hi @acsonefho, I have done a full testing and found a couple of issues. I have created a PR to your branch addressing them, could you have a look? https://github.com/acsone/storage/pull/1

acsonefho committed:
Merge pull request #1 from ForgeFlow/13.0-storage_backend_ftp-ff

[FIX] storage_backend_ftp: fix list and get

LoisRForgeFlow reviewed (approved):
LGTM

simahawk commented on storage_backend_ftp/components/ftp_adapter.py:
```diff
@@ -0,0 +1,148 @@
+        # Due to a bug into between ftplib and ssl, this part (about ssl) might not work!
+        # https://bugs.python.org/issue31727
+        security = None
+        if backend.ftp_security == "tls":
```
pls use a dict :wink:

simahawk commented on storage_backend_ftp/tests/test_ftp.py:
```diff
@@ -0,0 +1,159 @@
+        )
+
+    # Do not patch the entire ftplib otherwise the error_perm Exception
+    # become also a mock and then a traceback is genrated on the "except ftplib.error_perm"
```
```suggestion
    # become also a mock and then a traceback is generated on the "except ftplib.error_perm"
```

simahawk commented on storage_backend_sftp/models/storage_backend.py:
```diff
@@ -12,15 +12,17 @@ class StorageBackend(models.Model):
+    sftp_port = fields.Integer(string="SFTP Port", default=22)
     sftp_auth_method = fields.Selection(
-        string="Authentification Method",
+        string="SFTP Authentification Method",
```
```suggestion
        string="SFTP Authentication Method",
```

simahawk reviewed (approved):
LG

simahawk commented:
build is red

LoisRForgeFlow commented:
@acsonefho could you attend the final comments from @simahawk ?

LoisRForgeFlow commented:
I have superseded this PR due to inactivity here https://github.com/OCA/storage/pull/134, I think we can close this one.
